src/thread.cpp

```diff
@@ -124,8 +124,9 @@ Thread::Thread(unsigned int logicalCpu, const Func& func) {
   MARL_ASSERT(size > 0,
               "InitializeProcThreadAttributeList() did not give a size");
 
+  std::vector<uint8_t> buffer(size);
   LPPROC_THREAD_ATTRIBUTE_LIST attributes =
-      reinterpret_cast<LPPROC_THREAD_ATTRIBUTE_LIST>(alloca(size));
+      reinterpret_cast<LPPROC_THREAD_ATTRIBUTE_LIST>(buffer.data());
   CHECK_WIN32(InitializeProcThreadAttributeList(attributes, 1, 0, &size));
   defer(DeleteProcThreadAttributeList(attributes));
```

Comment:
This doesn't cause double delete, right?

Reply:
I do not believe so. The docs on this stuff are weak for microsoft's standards:

[`InitializeProcThreadAttributeList()`](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-initializeprocthreadattributelist) is handed a list which the caller has to allocate (this is the typical call-once with nullptr, get size, call again with buffer at least as big as reported size pattern).
[`DeleteProcThreadAttributeList()`](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-deleteprocthreadattributelist) says:

  _"Deletes the specified list of attributes for process and thread creation.
  `lpAttributeList` - The attribute list. This list is created by the `InitializeProcThreadAttributeList` function."_

I believe created != allocated, as `InitializeProcThreadAttributeList` does not allocate `lpAttributeList`.